spinoso-string/src/enc/ascii/mod.rs

```diff
@@ -198,7 +198,15 @@ impl AsciiString {
     pub fn get_char_slice(&self, range: Range<usize>) -> Option<&'_ [u8]> {
         let Range { start, end } = range;
 
-        self.inner.get(start..end).or_else(|| self.inner.get(start..))
```

Comment:
That's surely a better approach! I added unit tests for `Ascii` and `Binary` in 1909d91dfbdc1e9bf2192df3932799620643e474 but it looks like there is one edge case not considered by the current implementation, that is, `"abc"[2..1]` would be `"c"` instead of `""`. I added 9ceb37e800dbaa4aebd7a1045bb6c4b91249b21e to address this case.

Reply:
awesome and thanks for catching that bug! Glad to have the added tests.